### ISSUE TITLE
test_operator_gpu.test_correlation fails

### ISSUE THREAD

**Issue by marcoabreu** [apache/incubator-mxnet#9553]
http://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/incubator-mxnet/detail/master/284/pipeline

Job: Python3: MKLML-GPU
Command: ```tests/ci_build/ci_build.sh gpu_mklml PYTHONPATH=./python/ nosetests-3.4 --with-timer --verbose tests/python/gpu```
File containing test: https://github.com/apache/incubator-mxnet/blob/master/tests/python/unittest/test_operator.py

```
======================================================================

FAIL: test_operator_gpu.test_correlation

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/usr/local/lib/python3.5/dist-packages/nose/case.py", line 198, in runTest

    self.test(*self.arg)

  File "/usr/local/lib/python3.5/dist-packages/nose/util.py", line 620, in newfunc

    return func(*arg, **kw)

  File "/workspace/tests/python/gpu/../unittest/test_operator.py", line 2112, in test_correlation

    unittest_correlation((5,1,4,4), kernel_size = 3,max_displacement = 1,stride1 = 2,stride2 = 1,pad_size = 2,is_multiply = False)

  File "/workspace/tests/python/gpu/../unittest/test_operator.py", line 2101, in unittest_correlation

    assert_almost_equal(exe1.grad_dict['img1'].asnumpy(), grad1, rtol=1e-3, atol=1e-4)

  File "/workspace/python/mxnet/test_utils.py", line 495, in assert_almost_equal

    raise AssertionError(msg)

AssertionError: 

Items are not equal:

Error 268.456376 exceeds tolerance rtol=0.001000, atol=0.000100.  Location of maximum error:(4, 0, 0, 1), a=2.000000, b=1.555556

 a: array([[[[ 1.        ,  0.22222222,  0.77777779,  0.55555558],

         [ 2.        ,  3.11111116,  1.55555558, -0.66666669],

         [ 0.77777779,  1.55555558,  0.55555558,  0.55555558],...

 b: array([[[[ 1.        ,  0.22222222,  0.77777778,  0.55555556],

         [ 2.        ,  3.11111111,  1.55555556, -0.66666667],

         [ 0.77777778,  1.55555556,  0.55555556,  0.55555556],...
```

**Comment by marcoabreu** [comment]
http://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/incubator-mxnet/detail/PR-9560/1/pipeline

**Comment by marcoabreu** [comment]
http://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/incubator-mxnet/detail/PR-9522/1/pipeline

**Comment by anirudh2290** [comment]
are there any other instances of this failure apart from #9560 ?

**Comment by marcoabreu** [comment]
It also failed on the master branch: http://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/incubator-mxnet/detail/master/284/pipeline

**Comment by eric-haibin-lin** [comment]
closed via #9581

**Comment by ThomasDelteil** [comment]
Seems to be flaky again:
```
======================================================================

FAIL: test_operator_gpu.test_correlation

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/usr/local/lib/python3.5/dist-packages/nose/case.py", line 198, in runTest

    self.test(*self.arg)

  File "/usr/local/lib/python3.5/dist-packages/nose/util.py", line 620, in newfunc

    return func(*arg, **kw)

  File "/work/mxnet/tests/python/gpu/../unittest/common.py", line 157, in test_new

    orig_test(*args, **kwargs)

  File "/work/mxnet/tests/python/gpu/../unittest/test_operator.py", line 2379, in test_correlation

    unittest_correlation((5,1,15,15), kernel_size = 1,max_displacement = 5,stride1 = 1,stride2 = 1,pad_size = 5,is_multiply = False, dtype = dtype)

  File "/work/mxnet/tests/python/gpu/../unittest/test_operator.py", line 2349, in unittest_correlation

    assert_almost_equal(exe1.grad_dict['img1'].asnumpy(), grad1, rtol=1e-3, atol=1e-4)

  File "/work/mxnet/python/mxnet/test_utils.py", line 493, in assert_almost_equal

    raise AssertionError(msg)

AssertionError: 

Items are not equal:

Error 32.733224 exceeds tolerance rtol=0.001000, atol=0.000100.  Location of maximum error:(4, 0, 10, 11), a=63.000000, b=61.000000

 a: array([[[[ 57.,  97.,  79., ...,  63.,  65.,  95.],

         [ 49.,  67.,  45., ...,  51.,  51.,  79.],

         [ 73., 101., 121., ...,  57.,  91., 115.],...

 b: array([[[[ 57.,  97.,  79., ...,  63.,  65.,  95.],

         [ 49.,  67.,  45., ...,  51.,  51.,  79.],

         [ 73., 101., 121., ...,  57.,  91., 115.],...

-------------------- >> begin captured logging << --------------------

common: INFO: Setting test np/mx/python random seeds, use MXNET_TEST_SEED=1344567771 to reproduce.

--------------------- >> end captured logging << ---------------------
```
